app/src/main/java/com/samco/trackandgraph/statistics/Statistics.kt

```diff
@@ -557,3 +558,139 @@ internal fun getLargestBin(bins: List<List<Double>>?): Double? {
         ?.max()
 }
 
+
+/**
+ * Code to calculate nice y-intervals
+ */
+// There are two possible interval classes. The reason is that we want to make a large amount of
+// objects of the first class, which then will get filtered. Then we only have to do the more complex
+// calculations for the actually reasonable intervals
+data class PossibleIntervalProto (val interval: Double, val preferred: Boolean,
+                                  val base: Double)
+data class PossibleInterval     (val interval: Double, val preferred: Boolean,
+                                 val n_lines: Int    , val percentage_range_used: Double,
+                                 val bounds_min: Double, val bounds_max: Double,
+                                 val base: Double
+                                 )
+
+data class YAxisParameters (    val step_mode: StepMode, val n_intervals: Double,
+                                val bounds_min: Double, val bounds_max: Double,)
+
+fun finishProto (proto: PossibleIntervalProto, y_min: Double, y_max: Double, fixedBounds: Boolean): PossibleInterval {
```

Comment:
Some of these functions look like they could be private to me. I would appreciate if you could change this because until this file is refactored it helps to know which functions are actually called by other classes.